### PR TITLE
Add `num_traits` proc_macro helper for explicit import

### LINKED ISSUE #34: Cannot reexport `num-derive` proc_macros
Since the proc_macro implementation explicity declares an `extern num_traits` crate, any crate that wants to use it must also depend on `num-traits`.

https://github.com/rust-num/num-derive/blob/1b361d7af828467aa9a1d80d609793f0a312aae5/src/lib.rs#L80

This can be a problem in case a crate reexport a `num-derive` proc macros because it forces the user to also depend on `num-traits` otherwise it produces: `error[E0463]: can't find crate for `_num_traits`.

Would it be possible to add a feature flag that disable this behavior? Specifically it would be the user's responsability to make sure that the `num-traits` crate is in scope.

### PR DESCRIPTION
**Pull request by jean-airoldie**
This allows the user to specify a identifier for the `num_traits` crate
so that the proc_macros can directly depend in it.

This is usefull when reexporting `num-derive` and using `num-trait` as
a transitive import.

Fixes https://github.com/rust-num/num-derive/issues/34.

CC. @vglavnyy

### CODE REVIEW

File: tests/explicit_import.rs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#[macro_use]
+extern crate num_derive;
+extern crate num_traits as panini;
```
Comment (cuviper) [inline tests/explicit_import.rs]: Were you hungry when you wrote this? :slightly_smiling_face:

**Review by cuviper — COMMENTED** [review]
This looks nice, thank you.

One request though -- I think we need a test where `num-traits` is really not a direct dependency. The `panini` test is a start, but the real `num-traits` will still be in the crate prelude, so we don't _really_ know that it did the right thing. Maybe you could add another test crate like the `check/` path (used for no-std), and in this case depend on the `num` crate with `#[num_traits = "num"]`.

File: tests/explicit_import.rs (hunk 1 above)
Comment (jean-airoldie) [reply tests/explicit_import.rs]: Yeah that's probably why I was lazy and didn't make a test crate haha.

**Comment by jean-airoldie** [comment]
Also I was wondering about the format of the helper attribute. We could also use an helper a la serde: `#[num(crate = "import")]`, but at this points its more a matter of taste (unless some helpers are added in the future).

**Comment by cuviper** [comment]
Hi, sorry I forgot to finish review on this. I went ahead and tinkered with it myself a bit more, and I think this is good to go.

Thanks again!

bors r+

**Pull request merged**